emsdk.py

```diff
@@ -555,7 +556,10 @@ def unzip(source_filename, dest_dir, unpack_even_if_exists=False):
 
       unzip_to_dir = dest_dir
       if common_subdir:
-        unzip_to_dir = os.path.join('/'.join(dest_dir.split('/')[:-1]), 'unzip_temp')
+        if WINDOWS:
+          unzip_to_dir = os.path.join(os.getenv('TEMP'), 'unzip_temp')
+        else:
+          unzip_to_dir = os.path.join('/'.join(dest_dir.split('/')[:-1]), 'unzip_temp')
```

Comment:
What did `unzip_to_dir` looks like when it was causing issues for you?

Reply:
`\\?\C:unzip_temp\` which is inaccessible due the missing/stripped slash, as we were discussing in the issue ticket.

Reply:
But isn't the bug in `fix_potentially_long_windows_pathname`?

I mean at this point in the program `os.getenv('TEMP')` isn't in the `\\?\` form is it?

I mean if you print `os.path.join(os.getenv('TEMP'), 'unzip_temp') ` you don't see the problematic path do you?

Reply:
I would log the input and output of `fix_potentially_long_windows_pathname` because it seems like the bug must be in there no?

Reply:
That's what I thought initially, that the issue was from the path(s) constructed by `fix_potentially_long_windows_pathname`, but after doing a simple test as pointed in https://github.com/emscripten-core/emsdk/issues/412#issuecomment-582510880 i then realized what the actual/real problem was, the stripped out slash - and you helped me to reach this correct conclusion :-)

Reply:
So is the problem that you $TEMP env var doesn't have a slash?   Is it just "C:something"?   I guess fix_potentially_long_windows_pathname should be updated to handle that case?

Reply:
No, the problem was with the former `dest_dir.split('/')[:-1]` 

To show you, the below is what i get with this patch applied:

```diff
@@ -554,8 +555,15 @@ def unzip(source_filename, dest_dir, unpack_even_if_exists=False):
           break

       unzip_to_dir = dest_dir
+      print('dest_dir = ' + dest_dir)
+      print("dest_dir.split('/')[:-1] = " + ''.join(dest_dir.split('/')[:-1]))
+      print("'/'.join(dest_dir.split('/')[:-1]) = " + '/'.join(dest_dir.split('/')[:-1]))
+      sys.exit(1)
       if common_subdir:
-        unzip_to_dir = os.path.join('/'.join(dest_dir.split('/')[:-1]), 'unzip_temp')
+        if WINDOWS:
+          unzip_to_dir = os.path.join(os.getenv('TEMP'), 'unzip_temp')
+        else:
+          unzip_to_dir = os.path.join('/'.join(dest_dir.split('/')[:-1]), 'unzip_temp')
```

The output:

```bash
$ EMSDK_VERBOSE=1 emsdk update
download_and_unzip(zipfile=https://github.com/emscripten-core/emsdk/archive/master.zip, dest_dir=C:/emsdk)
download_file(url=https://github.com/emscripten-core/emsdk/archive/master.zip, dstpath=zips/)
mkdir_p(C:/emsdk/zips)
Downloading: C:/emsdk/zips/master.zip from https://github.com/emscripten-core/emsdk/archive/master.zip, 58641 Bytes
unzip(source_filename=C:/emsdk/zips/master.zip, dest_dir=C:/emsdk)
Unpacking 'C:/emsdk/zips/master.zip' to 'C:/emsdk'
mkdir_p(C:/emsdk)

dest_dir = C:/emsdk
dest_dir.split('/')[:-1] = C:
'/'.join(dest_dir.split('/')[:-1]) = C:
```

So, the original `dest_dir` was `C:/emsdk`, and after the call to `os.path.join(....)` then `unzip_to_dir` did turn `C:unzip_temp` which is not correct.

Later on, `fix_potentially_long_windows_pathname` converts it to `\\?\C:unzip_temp\` (so it is just correctly prepending the `\\?\` prefix but the problem is not on this)

There, i could also have fixed the issue by avoiding stripping the slash with `dest_dir.split('/')[:-1]`, but i thought more convenient just using $TEMP as the target where to extract the files.

Reply:
> \\?\C:unzip_temp\ which is inaccessible

I just realized i wrongly answered to your first question, sorry - hopefully my last post does clarify everything.

Next i need to grab some sleep 😅

Reply:
I was a little afraid of doing such kind of change that could affect non-Windows platforms, but given you seem confident about it i have just made your suggested change. 👍